### PR TITLE
enh(resources): add basic extended filters

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
**description**:  update interface of criterias (basic criterias and extended criteria)
**more details**: https://centreon.atlassian.net/jira/software/c/projects/MON/boards/211?assignee=61d809797c6f9800706f84a7&selectedIssue=MON-20803
this PR regroup those tickets :  https://centreon.atlassian.net/browse/MON-20803 (basic criteria) , https://centreon.atlassian.net/browse/MON-20804 (extended criteria)

**screens**: (check snapshots)

**video**

https://github.com/centreon/centreon/assets/97687698/36b304b0-5927-4f97-871e-68cfea401cb1

NB: lighthouse (pour mesurer less performances) sera inclut dans un ticket separé

### CODE REVIEW

File: centreon/package.json [hunk 1]
```diff
@@ -148,6 +148,8 @@
     "i18next": "^22.0.4",
     "jotai": "2.0.4",
     "jotai-suspense": "^0.1.0",
+    "lodash": "^4.17.21",
```
Comment (bdauria) [inline centreon/package.json:151]: don't use lodash but rather a ramda alternative: https://ramdajs.com/

File: centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/extendedFilter/FilterSearch/useDebounce.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { useEffect, useMemo, useRef } from 'react';
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/extendedFilter/FilterSearch/useDebounce.ts:1]: There is already a useDebounce hook here: https://github.com/centreon/centreon/blob/develop/centreon/packages/ui/src/utils/useDebounce.ts

File: centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/InputGroupe.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import { useState } from 'react';
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/InputGroupe.tsx:1]: file should be named InputGroup.tsx :)

File: centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/atoms.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { atomWithStorage } from 'jotai/utils';
+import { atom } from 'jotai';
+
+import { SelectedResourceType } from '../model';
+
+export const selectedStatusByResourceTypeAtom =
+  atomWithStorage<Array<SelectedResourceType> | null>(
+    'FilterSelectedStatus',
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/atoms.ts:8]: I think you should contextualize the key (for instance: centreon-{{version}}-resourceStatusRevamp-filterSelecedStatus

File: centreon/www/front_src/src/Resources/Filter/index.tsx [hunk 5]
```diff
@@ -85,30 +81,48 @@ import {
...
+        title={t(labelClearFilter) as string}
+        onClick={onClear}
+      >
+        <CloseIcon color="action" fontSize="small" />
+      </IconButton>
+      {/* <Suspense
+        fallback={<LoadingSkeleton height={24} variant="circular" width={24} />}
+      >
+        <Criterias />
+      </Suspense> */}
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/index.tsx:112]: To be removed

File: centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/utils.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+          return [item[key]?.toString(), item];
+        }
+
+        return [key, item];
+      })
+    ).values()
+  ];
+};
+
+export const sort = ({
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/utils.ts:141]: Why haven't you use the sort function from Ramda ?

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 7]
```diff
@@ -328,28 +599,171 @@ describe('Custom filters', () => {
...
+    cy.makeSnapshot();
+  });
+
+  BasicCriteriasParams.forEach(([label, data]) => {
+    const searchValue = 'foobar';
+
+    data.forEach((element) => {
+      const { criteria, value, type, endpointParam, requestToWait } = element;
+
+      it(`executes a listing request with current search and selected ${criteria} criteria value when it's changed:${label}`, () => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:737]:
```suggestion
      it(`executes a listing request with current search and selected ${criteria} criteria value when ${label} has changed`, () => {
```

File: centreon/www/front_src/src/Resources/Filter/Criterias/SaveActions.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    setIsCreateFilter(false);
+  };
+
+  const cancelUpdateFilter = (): void => {
+    setIsUpdateFilter(false);
+  };
+
+  return (
+    <>
+      {isCreateFilter && (
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Filter/Criterias/SaveActions.tsx:68]:
```suggestion
      {isCreatingFilter && (
```

File: centreon/www/front_src/src/Resources/Filter/Criterias/SaveActions.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        <CreateFilterDialog
+          callbackSuccess={createFilterCallback}
+          open={isCreateFilter}
+          payloadAction={{ criterias: currentFilter?.criterias }}
+          request={createFilter}
+          onCancel={cancelCreateFilter}
+        />
+      )}
+
+      {isUpdateFilter && (
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Filter/Criterias/SaveActions.tsx:78]:
```suggestion
      {isUpdatingFilter && (
```

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -5901,7 +5901,7 @@ msgstr "Monitoring Engine Konfigurationsdatei anzeigen"
 #: centreon/www/include/configuration/configObject/service_dependency/formServiceDependency.php:256
 #: centreon/www/include/configuration/configObject/hostgroup_dependency/formHostGroupDependency.php:102
 msgid "Information"
-msgstr "Information"
+msgstr "Informatie"
```
[the remark this answers is not in the document]
Comment (Noha-ElAbrouki) [reply centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po]: yess Cecil fixed for me

File: centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -16032,6 +16032,9 @@ msgstr ""
 #  msgid "View by service"
 #  msgstr ""
 
+msgid "Advanced mode"
+msgstr "Geavanceerde modus"
```
[the remark this answers is not in the document]
Comment (Noha-ElAbrouki) [reply centreon/lang/de_DE.UTF-8/LC_MESSAGES/messages.po]: for a long time i thought dutsch = german :s

**Review by callapa — DISMISSED** [review]

File: centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/atoms.ts [hunk 12]
```diff
@@ -0,0 +1,13 @@
+import { atomWithStorage } from 'jotai/utils';
+import { atom } from 'jotai';
+
+import { SelectedResourceType } from '../model';
+
+export const selectedStatusByResourceTypeAtom =
+  atomWithStorage<Array<SelectedResourceType> | null>(
+    'centreon-23.10-resourceStatusRevamp-filterSelectedStatus',
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/atoms.ts:8]: To do in another PR: We must find a way to get the set version using the platform version (23.10, 24.4) dynamically

**Review by Thebarda — DISMISSED** [review]

File: centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/atoms.ts (hunk 12 above)
Comment (Noha-ElAbrouki) [reply centreon/www/front_src/src/Resources/Filter/criteriasNewInterface/basicFilter/atoms.ts:8]: ook noté

**Review by cg-tw — DISMISSED** [review]

**Review by callapa — DISMISSED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by bdauria — DISMISSED** [review]

**Review by Thebarda — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Review by dmyios — APPROVED** [review]
